crates/rolldown/src/module_loader/module_loader.rs

```diff
@@ -82,8 +84,29 @@ impl ModuleLoader {
 
     let tx_to_runtime_module = tx.clone();
 
-    let common_data =
-      Arc::new(TaskContext { options: Arc::clone(&options), tx, resolver, fs, plugin_driver });
+    let common_data = Arc::new(TaskContext {
+      options: Arc::clone(&options),
+      tx,
+      resolver,
+      fs,
+      plugin_driver,
+      meta: TaskContextMeta {
+        replace_global_define_config: if !options.define.is_empty() {
```

Comment:
I guess you could just put `ReplaceGlobalDefinesConfig` in the `options.define`.

Reply:
The only thing I am hesitated to do that is if we want the raw `NormalizedBundlerOptions`. For now the `NormalizedBundlerOptions` is almost `1:1` for `rolldown.config.js`.

Reply:
No need worried about that. The purpose of `NormalizedBundlerOptions` is make we happy, no need to dealing with something raw. However, the current implementation of the PR is also fine.

Reply:
Ok, then I will replace it.

Reply:
If we  replace `define` with `ReplaceGlobalDefinesConfig`(but we need handle error and propagate errors), which will lead change all upstream function return types to `Result<T>`. let's merge it first.